container/src/main/java/org/wildfly/swarm/internal/SystemDependencyResolution.java

```diff
@@ -0,0 +1,144 @@
+package org.wildfly.swarm.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.importer.ExplodedImporter;
+import org.jboss.shrinkwrap.api.importer.ZipImporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.wildfly.swarm.bootstrap.util.WildFlySwarmClasspathConf;
+
+/**
+ * @author Heiko Braun
+ * @since 18/07/16
+ */
+public class SystemDependencyResolution implements DependencyResolution {
+
+    public SystemDependencyResolution(List<String> classpath, String pwd, String javaHome, List<String> testClasspath) {
+        this.classpath = classpath;
+        this.pwd = pwd;
+        this.javaHome = javaHome;
+        this.testClasspath = testClasspath;
+    }
+
+    public SystemDependencyResolution setDryRun(boolean b) {
+        this.dryRun = b;
+        return this;
+    }
+
+    @Override
+    public Map<String, JavaArchive> resolve(List<String> exclusions) throws IOException {
+
+        final Map<String, JavaArchive> archives = new HashMap<>();
+
+        exclusions.replaceAll(s -> s.replace('.', File.separatorChar));
+
+        if (classpath != null) {
+            WildFlySwarmClasspathConf classpathConf = new WildFlySwarmClasspathConf();
+            Set<String> classpathElements = new HashSet<>();
+            Set<String> providedGAVs = new HashSet<>();
+            List<String> testClasspathElements = testClasspath!=null ?
+                    testClasspath : Collections.EMPTY_LIST;
+
+            // filter classpath entries, makes some assumptions about maven
+            // TODO: not sure how this actually works with gradle. For now I assume gradle works on the same same local repository ...
+
+            final String m2Repo = "m2" + File.separator + "repository";
```

Comment:
right, so the default is `${user.home}/.m2/repository`, but i t can be overriden in either 

a) `settings.xml`
b) or using `-Dmaven.repo.local`

But considering that the algo so far is at best an approximation, I think a solution that works for the majority of users would be good enough. So one way to approach it would be to assume the majority uses the default location and still respect `-Dmaven.repo.local`, if given. I think the only case this would not cover is changes to `settings.xml`.